pkg/endpoint/local_ip.go

```diff
@@ -44,21 +44,38 @@ func getLocalIPFromRoutes(family k8snet.IPFamily) (string, error) {
 	}
 
 	for i := range routes {
-		if routes[i].Gw != nil {
-			return routes[i].Src.String(), nil
+		if routes[i].Gw != nil && routes[i].Src != nil {
+			ip := routes[i].Src
+			if family == k8snet.IPv6 && !isValidGlobalIPv6(ip) {
```

Comment:
Maybe we  should add a comment here to explain why  do we check that IPv6 address is valid .

    OVNK configures special internal masquerade IPs,
    these IPs are used internally by OVNK infra dataplane, and they should
    not be selected as private IPs. [1]
 
    [1] https://github.com/ovn-kubernetes/ovn-kubernetes/blob/eea781bd80d4afb29e0d43c81ea5503fbe452d78/docs/design/host-to-node-port-hairpin-trafficflow.md?plain=1#L59

Reply:
Done.